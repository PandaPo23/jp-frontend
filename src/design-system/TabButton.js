import React from 'react';

import targetedProps from '../utils/targetedProps';
import Button from './Button';

const defaults = {
  borderRadius: 'top.sm',
  borderColor: 'transparent',
  borderTop: 0,
  borderBottom: 4,
  borderLeft: 0,
  borderRight: 0,
  boxShadow: 0,
  bold: false,
  uppercase: true,
  bg: 'surface',
  px: 3,
  py: 3,
  minWidth: 'auto',
  color: 'misc.muted',
  letterSpacing: 'tracked',
};

const activeDefaults = {
  borderColor: 'secondary',
  bold: true,
  color: 'on.background',
};

const TabButton = ({ active = false, children, ...props }) => {
  const activeProps = { ...activeDefaults, ...targetedProps(props, 'active') };
  return (
    <Button {...defaults} {...(active && activeProps)} {...props}>
      {children}
    </Button>
  );
};

export default TabButton;
