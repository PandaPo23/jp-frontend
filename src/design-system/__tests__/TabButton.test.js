import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import TabButton from '../TabButton';

describe('TabButton component', () => {
  it('renders non-active TabButton correctly', () => {
    const localProps = {
      active: false,
      onClick: () => {},
      children: 'Normal Tab',
    };
    const wrapper = mount(<TabButton {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders active TabButton correctly', () => {
    const localProps = {
      active: true,
      onClick: () => {},
      children: 'Active Tab',
    };
    const wrapper = mount(<TabButton {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('triggers onClick event handler prop when button is clicked', () => {
    const localProps = {
      active: false,
      onClick: jest.fn(),
      children: 'Normal Tab',
    };
    const wrapper = mount(<TabButton {...localProps} />);
    wrapper.find('button').simulate('click');
    expect(localProps.onClick).toHaveBeenCalled();
  });
});
