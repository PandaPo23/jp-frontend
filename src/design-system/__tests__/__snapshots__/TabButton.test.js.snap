// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton component renders active TabButton correctly 1`] = `
.emotion-0 {
  color: on.background;
  background-color: surface;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-letter-spacing: tracked;
  -moz-letter-spacing: tracked;
  -ms-letter-spacing: tracked;
  letter-spacing: tracked;
  display: inline-block;
  min-width: auto;
  border: 1px;
  border-bottom: 4px;
  border-color: secondary;
  border-left: 0;
  border-radius: top.sm;
  border-right: 0;
  border-top: 0;
  box-shadow: 0;
  cursor: pointer;
  font-weight: bold;
  text-transform: uppercase;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<TabButton
  active={true}
  onClick={[Function]}
>
  <Button
    as="button"
    bg="surface"
    bold={true}
    border={1}
    borderBottom={4}
    borderColor="secondary"
    borderLeft={0}
    borderRadius="top.sm"
    borderRight={0}
    borderTop={0}
    boxShadow={0}
    color="on.background"
    display="inline-block"
    hoverable={true}
    letterSpacing="tracked"
    minWidth="auto"
    onClick={[Function]}
    px={3}
    py={3}
    uppercase={true}
  >
    <button
      className="emotion-0"
      onClick={[Function]}
    >
      Active Tab
    </button>
  </Button>
</TabButton>
`;

exports[`TabButton component renders non-active TabButton correctly 1`] = `
.emotion-0 {
  color: misc.muted;
  background-color: surface;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-letter-spacing: tracked;
  -moz-letter-spacing: tracked;
  -ms-letter-spacing: tracked;
  letter-spacing: tracked;
  display: inline-block;
  min-width: auto;
  border: 1px;
  border-bottom: 4px;
  border-color: transparent;
  border-left: 0;
  border-radius: top.sm;
  border-right: 0;
  border-top: 0;
  box-shadow: 0;
  cursor: pointer;
  text-transform: uppercase;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<TabButton
  active={false}
  onClick={[Function]}
>
  <Button
    as="button"
    bg="surface"
    bold={false}
    border={1}
    borderBottom={4}
    borderColor="transparent"
    borderLeft={0}
    borderRadius="top.sm"
    borderRight={0}
    borderTop={0}
    boxShadow={0}
    color="misc.muted"
    display="inline-block"
    hoverable={true}
    letterSpacing="tracked"
    minWidth="auto"
    onClick={[Function]}
    px={3}
    py={3}
    uppercase={true}
  >
    <button
      className="emotion-0"
      onClick={[Function]}
    >
      Normal Tab
    </button>
  </Button>
</TabButton>
`;
